Framework/Algorithms/src/CompareWorkspaces.cpp

```diff
@@ -746,16 +748,18 @@ bool CompareWorkspaces::checkInstrument(API::MatrixWorkspace_const_sptr ws1,
     return false;
   }
 
-  const Geometry::ParameterMap &ws1_parmap = ws1->constInstrumentParameters();
-  const Geometry::ParameterMap &ws2_parmap = ws2->constInstrumentParameters();
+  if (getProperty("CheckParameterMap")) {
+    const Geometry::ParameterMap &ws1_parmap = ws1->constInstrumentParameters();
+    const Geometry::ParameterMap &ws2_parmap = ws2->constInstrumentParameters();
 
-  if (ws1_parmap != ws2_parmap) {
-    g_log.debug()
-        << "Here information to help understand parameter map differences:\n";
-    g_log.debug() << ws1_parmap.diff(ws2_parmap);
-    recordMismatch(
-        "Instrument ParameterMap mismatch (differences in ordering ignored)");
-    return false;
+    if (ws1_parmap != ws2_parmap) {
+      g_log.debug()
+          << "Here information to help understand parameter map differences:\n";
+      g_log.debug() << ws1_parmap.diff(ws2_parmap);
```

Comment:
Comparing a whole parameter map is a big task in come cases. Any idea what the actual complexity of this is? Have you tried timing this comparison with logs in debug for two workspaces containing large parameter maps? If this takes any longer that 0.5 seconds in the worst cases, you may need to consider adding progress updating as part of this.